tests/func/test_checkout.py

```diff
@@ -581,3 +582,24 @@ def test_checkout_no_checksum(repo_dir, dvc_repo):
     stage = dvc_repo.run(outs=[repo_dir.FOO], no_exec=True, cmd="somecmd")
     dvc_repo.checkout(stage.path, force=True)
     assert not os.path.exists(repo_dir.FOO)
+
+
+def test_checkout_should_avoid_copy_for_existing_cache(
```

Comment:
It feels like it would be appropriate to test other scenarios too, since they are not trivial. E.g. if we have symlink/hardlink and cache_type is copy. And if we have copy and cache_type is symlink/hardlink. Maybe it is worth making into a unit test even instead. What do you think?

Reply:
Seems reasonable, ill do it